packages/core/lib/commands/call/run.js

```diff
@@ -0,0 +1,167 @@
+module.exports = async function (options) {
+  const fs = require("fs");
+  const util = require("util");
+  const { Environment } = require("@truffle/environment");
+  const OS = require("os");
+  const Codec = require("@truffle/codec");
+  const Encoder = require("@truffle/encoder");
+  const Decoder = require("@truffle/decoder");
+  const TruffleError = require("@truffle/error");
+  const { fetchAndCompile } = require("@truffle/fetch-and-compile");
+  const loadConfig = require("../../loadConfig");
+
+  if (options.url && options.network) {
+    const message =
+      "" +
+      "Mutually exclusive options, --url and --network detected!" +
+      OS.EOL +
+      "Please use either --url or --network and try again." +
+      OS.EOL +
+      "See: https://trufflesuite.com/docs/truffle/reference/truffle-commands/#call" +
+      OS.EOL;
+    throw new TruffleError(message);
+  }
+
+  let config = loadConfig(options);
+  await Environment.detect(config);
+
+  const [contractNameOrAddress, functionNameOrSignature, ...args] = config._;
+  let functionEntry, transaction;
+
+  const { encoder, decoder } = config.fetchExternal
+    ? await sourceFromExternal(contractNameOrAddress, config)
+    : await sourceFromLocal(contractNameOrAddress, config);
+
+  ({ abi: functionEntry, tx: transaction } = await encoder.encodeTransaction(
```

Comment:
Thanks for mentioning this @haltman-at . I didn't have much knowledge about this. This is really cool that encoder handles these things too. :)

Reply:
Ah yeah sorry I guess you weren't following the initial discussions between me and @gnidan about `truffle call`.  The thing is that Truffle Encoder was originally meant to support Truffle Contract, and so I made it with that in mind -- the idea that what would be passed in would be JS objects, rather than necessarily strings.  This led to two problems when considering Truffle Call:

1. There was no way to specify complex types like arrays or structs; thus the `allowJson` option had to be added
2. For compatibility with ethers, *any* input was allowed as a boolean.  On the command line there's no need for that and it would only cause confusion, thus, `strictBooleans`.

Based on the discussion of addresses below, possibly I might want to add another new option for recognizing contract names... well, we can discuss this.

Reply:
Thanks for explaining this, @haltman-at.